oxidation/libparsec/crates/types/tests/test_manifest.rs

```diff
@@ -101,6 +101,8 @@ fn serde_file_manifest(alice: &Device) {
         &alice.verify_key(),
         &alice.device_id,
         now,
+        None,
```

Comment:
Can you add a separate tests to check that the `expected_*` fields works as expected ?

something like:

```rs
fn test_file_manifest_expected_checks() {
  let manifest = FileManifest(...);
  let encrypted = manifest.dump_sign_and_encrypt(...)

  // Invalid expects

  assert_eq!(FileManifest.decrypt_verify_and_load(
    encrypted,
    bad_author,
    expected_timestamp,
    None,
    None,
  ), Err(...))

  assert_eq!(FileManifest.decrypt_verify_and_load(
    encrypted,
    expected_author,
    bad_timestamp,
    None,
    None,
  ), Err(...))

  assert_eq!(FileManifest.decrypt_verify_and_load(
    encrypted,
    expected_author,
    expected_timestamp,
    bad_id,
    None,
  ), Err(...))

  assert_eq!(FileManifest.decrypt_verify_and_load(
    encrypted,
    expected_author,
    expected_timestamp,
    expected_id,
    bad_version,
  ), Err(...))

  // Sanity check with good expects

  FileManifest.decrypt_verify_and_load(
    encrypted,
    expected_author,
    expected_timestamp,
    None,
    None,
  ).unwrap()

  FileManifest.decrypt_verify_and_load(
    encrypted,
    expected_author,
    expected_timestamp,
    expected_id,
    expected_version,
  ).unwrap()
}
```

Reply:
I would add a test for `verify` that would also test `decrypt_verify_and_load`

Reply:
done